Mathlib/Geometry/Euclidean/Altitude.lean

```diff
@@ -0,0 +1,126 @@
+/-
+Copyright (c) 2020 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import Mathlib.Analysis.InnerProductSpace.Projection
+import Mathlib.LinearAlgebra.AffineSpace.FiniteDimensional
+
+/-!
+# Altitudes of a simplex
+
+This file defines the altitudes of a simplex.
+
+## Main definitions
+
+* `altitude` is the line that passes through a vertex of a simplex and
+  is orthogonal to the opposite face.
+
+## References
+
+* <https://en.wikipedia.org/wiki/Altitude_(triangle)>
+
+-/
+
+noncomputable section
+
+namespace Affine
+
+namespace Simplex
+
+open Finset AffineSubspace
+
+variable {V : Type*} {P : Type*} [NormedAddCommGroup V] [InnerProductSpace ℝ V] [MetricSpace P]
+  [NormedAddTorsor V P]
+
+/-- An altitude of a simplex is the line that passes through a vertex
+and is orthogonal to the opposite face. -/
+def altitude {n : ℕ} (s : Simplex ℝ P (n + 1)) (i : Fin (n + 2)) : AffineSubspace ℝ P :=
+  mk' (s.points i) (affineSpan ℝ (s.points '' ↑(univ.erase i))).directionᗮ ⊓
```

Comment:
I guess using `faceOpposite` here is future work, as this PR is just a move?

Reply:
Yes, any change to how `altitude` is defined would be for future work since this PR is only moving existing content.

bors r+